### PR TITLE
chore(setUserAgent): add initial util and types

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add initial `setUserAgent` util and types

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/utils/index.ts [hunk 1]
```diff
@@ -1,265 +1,3 @@
-import { Modifiers } from '../types';
```
Comment (calebpollman) [inline packages/ui/src/utils/index.ts:1]: Contents moved to `'./utils'`

File: packages/ui/src/utils/setUserAgent/constants.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  AuthAction,
+  Category,
+} from '@aws-amplify/core/internals/utils';
+
+// replace type once Category input types are available
+export const ACCOUNT_SETTINGS_INPUT_BASE: Omit<
+  SetCustomUserAgentInput,
+  'additionalDetails'
+> = {
+  apis: [AuthAction.DeleteUser, AuthAction.UpdatePassword],
```
Comment (calebpollman) [inline packages/ui/src/utils/setUserAgent/constants.ts:13]: These need to be audited in a follow up

File: packages/ui/src/utils/setUserAgent/setUserAgent.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  let input: SetCustomUserAgentInput | undefined;
+
+  const additionalDetails: SetCustomUserAgentInput['additionalDetails'] = [
+    [componentName],
+    [`ui-${packageName}/${version}`],
+  ];
+
+  switch (componentName) {
+    case 'AccountSettings': {
+      input = ACCOUNT_SETTINGS_INPUT_BASE as SetCustomUserAgentInput;
```
Comment (calebpollman) [inline packages/ui/src/utils/setUserAgent/setUserAgent.ts:64]:
```suggestion
      // remove cast when Category input types are available
      input = ACCOUNT_SETTINGS_INPUT_BASE as SetCustomUserAgentInput;
```

File: packages/ui/src/utils/setUserAgent/setUserAgent.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      break;
+    }
+    case 'InAppMessaging': {
+      input = IN_APP_MESSAGING_INPUT_BASE as SetCustomUserAgentInput;
```
Comment (calebpollman) [inline packages/ui/src/utils/setUserAgent/setUserAgent.ts:68]:
```suggestion
      // remove cast when Category input types are available
      input = IN_APP_MESSAGING_INPUT_BASE as SetCustomUserAgentInput;
```

File: packages/ui/src/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
+import { Modifiers } from '../types';
```
Comment (calebpollman) [inline packages/ui/src/utils/utils.ts:1]: No changes to this file

**Review by dbanksdesign — APPROVED** [review]

File: packages/ui/src/utils/setUserAgent/setUserAgent.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+export const setUserAgent = ({
+  componentName,
+  packageName,
+  version,
+}: SetUserAgentOptions): (() => void) => {
+  let input: SetCustomUserAgentInput | undefined;
+
+  const additionalDetails: SetCustomUserAgentInput['additionalDetails'] = [
+    [componentName],
+    [`ui-${packageName}/${version}`],
```
Comment (jimblanc) [inline packages/ui/src/utils/setUserAgent/setUserAgent.ts:59]:
```suggestion
    [`ui-${packageName}`, version],
```

I think you can just do this here, JS will handle formatting the tuple

File: packages/ui/src/utils/setUserAgent/setUserAgent.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  ];
+
+  switch (componentName) {
```
Comment (calebpollman) [inline packages/ui/src/utils/setUserAgent/setUserAgent.ts:62]: Exhaustive `componentName` constant inputs to be added in follow up

**Review by jimblanc — APPROVED** [review]
LGTM

File: packages/ui/src/utils/setUserAgent/setUserAgent.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  | 'react-native-auth'
+  | 'react-notifications'
+  | 'react-storage'
+  | 'vue';
+
+type ComponentName =
+  | 'AccountSettings'
+  | 'Authenticator'
+  | 'InAppMessaging'
+  | 'FaceLivenessDetector'
```
Comment (esauerbo) [inline packages/ui/src/utils/setUserAgent/setUserAgent.ts:27]: Is JS adding types for liveness?

**Pull request merged**